src/app.js

```diff
@@ -39,18 +39,84 @@ export function app() {
     innerText: 'Pokédex'
   });
 
+  // Search
+
+  const searchWrapper = createElement('div', {
+    className: 'searchWrapper'
+  });
+  const search = createElement('input', {
+    className: 'searchWrapper__input',
+    type: 'text'
+  });
+
+  // Search Results
+
+  const searchResultsWrapper = createElement('div', {
+    className: 'searchResultsWrapper'
+  });
+
+  async function createSearchResults() {
+    search.addEventListener('input', async event => {
+      const input = event.target;
+      const searchQuery = input.value;
+
+      // Get Data
+      const pokemonData = await fetchPokemons();
+
+      // Filter results
+      const filteredPokemons = filterResults(searchQuery, pokemonData);
+
+      // Create search results
+      removeAllChilds(searchResultsWrapper);
+
+      filteredPokemons.forEach(pokemon => {
+        const searchResult = createElement('div', {
+          className: 'searchResultsWrapper__searchResult'
+        });
+        searchResult.dataset.pokemonid = pokemon.id;
+        const searchResultTitle = createElement('h2', {
+          className: 'searchResultsWrapper__title',
+          innerText: pokemon.name
+        });
+        const searchResultImageWrapper = createElement('div', {
+          className: 'searchResultsWrapper__imageWrapper'
+        });
+        const searchResultImage = createElement('img', {
+          className: 'searchResultsWrapper__image',
+          src: pokemon.imagePath
+        });
+        appendElement(searchResultImageWrapper, [searchResultImage]);
+        appendElement(searchResult, [
+          searchResultTitle,
+          searchResultImageWrapper
+        ]);
+        appendElement(searchResultsWrapper, [searchResult]);
+      });
+    });
+
+    function filterResults(searchQuery, pokemonData) {
+      // Get Pokemon names
+
+      const filteredPokemons = [];
+      pokemonData.forEach(pokemon => {
```

Comment:
Instead of a `forEach`, you can profit from `filter`

Reply:
Solved with https://github.com/hauptdigital/redesigned-pokedex/pull/5/commits/d641847d7552e64c389fa0d0ef427a5ae3b04ab1